scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -785,6 +795,9 @@ private void processAlertRecord(final Record record, final Connection connection
 					errorHandler.onError(alert.getPeer(), alert.getLevel(), alert.getDescription());
```

Comment:
The ErrorHandler provides details about the received alert message, which will be only available for "HandshakeException" (containing a reference to the alert message). So with the above change to Throwable, a "instance of" check would be required.